src/common/users.ts

```diff
@@ -97,16 +104,50 @@ const rankFromProgress = (progress: number) => {
   return 0;
 };
 
+interface ReadingDaysArgs {
+  userId: string;
+  timezone?: string;
+  limit?: number;
+}
+
+const getUserReadingDays = (
+  con: Connection,
+  { userId, timezone = 'utc', limit = 8 }: ReadingDaysArgs,
+) => {
+  const now = `timezone('${timezone}', now())`;
+  return con.query(
+    `
+    with filtered_view as (
+      select *, CAST(v."timestamp" at time zone '${timezone}' AS DATE) as day
+      from "view" v
+      where "userId" = $1 and v."timestamp" at time zone '${timezone}' >= date_trunc('week', ${now})
```

Comment:
I think it's better to support a custom time frame here. This way we can reuse it for other purposes as well

Reply:
Agreed would be better to support the time frame, so we can use it on the profile page as well.